contracts/DappStakingPool.sol

```diff
@@ -313,6 +315,8 @@ contract DappStakingPool is OwnableUpgradeable, ITransferPositionCallback {
                 bntToken.transfer(msg.sender, networkAmount);
             }
         }
+
+        if(userInfo.amount == 0) userInfo.positionId = 0;
```

Comment:
This may work perfectly, although i was thinking
`if (newLpAmount == 0)`
They are equivalent though so we can leave it as you made it